dirty_cat/similarity_encoder.py

```diff
@@ -369,6 +401,31 @@ def transform(self, X, fast=True):
 
         """
 
+        if self.handle_missing not in ['error', '']:
+            template = ("handle_missing should be either 'error' or "
+                        "'', got %s")
+            raise ValueError(template % self.handle_missing)
+        if hasattr(X, 'iloc') and X.isna().values.any():
+            if self.handle_missing == 'error':
+                msg = ("Found missing values in input data.")
+                raise ValueError(msg)
+            if self.handle_missing != 'error':
+                X = X.fillna(self.handle_missing)
+        elif not hasattr(X, 'dtype') and isinstance(X, list):
+            X = np.asarray(X, dtype=object)
+
+        if hasattr(X, 'dtype'):
+            if np.issubdtype(X.dtype, np.str_):
+                X = np.asarray(X, dtype=object)
```

Comment:
If the dtype is a string, we don't have to worry about missing values: there cannot be any. So we should just skip ahead, rather than converting.

Reply:
Ah yes, I removed it from one function but I guess I missed the other.